e2e/app/element/base-element.ts

```diff
@@ -146,23 +146,25 @@ export default class BaseElement {
       while (Date.now() - startTime < 30 * 1000) {
         const viewport = await element.isIntersectingViewport();
         if (viewport) {
-          const box = await element.boundingBox();
-          const x = box.x + box.width / 2;
-          const y = box.y + box.height / 2;
-          if (previousX !== undefined && previousY !== undefined) {
-            // tslint:disable:triple-equals
-            if (
-              parseFloat(previousX.toFixed(7)) === parseFloat(x.toFixed(7)) &&
-              parseFloat(previousY.toFixed(7)) === parseFloat(y.toFixed(7))
-            ) {
-              break;
+          const boundingBox = await element.boundingBox();
+          const boxModel = await element.boxModel();
```

Comment:
`boundingBox` method returns the bounding box of the element (relative to the main frame), or null if the element is not visible.
`boxModel` method returns boxes of the element, or null if the element is not visible.